src/sentry/integrations/github_enterprise/webhook.py

```diff
@@ -0,0 +1,439 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+
+import dateutil.parser
+import hashlib
+import hmac
+import logging
+import six
+
+from django.db import IntegrityError, transaction
+from django.http import HttpResponse
+from django.utils.crypto import constant_time_compare
+from django.utils.decorators import method_decorator
+from django.views.decorators.csrf import csrf_exempt
+from django.views.generic import View
+from django.utils import timezone
+from simplejson import JSONDecodeError
+from sentry.models import (
+    Commit, CommitAuthor, CommitFileChange, Identity, Integration, PullRequest,
+    Repository, User
+)
+from sentry.utils import json
+from sentry.constants import ObjectStatus
+from sentry.integrations.exceptions import ApiError
+from .repository import GitHubEnterpriseRepositoryProvider
+from .client import GitHubEnterpriseAppsClient
+
+logger = logging.getLogger('sentry.webhooks')
+
+
+def is_anonymous_email(email):
+    return email[-25:] == '@users.noreply.github.com'
+
+
+def get_external_id(username):
+    return 'github:%s' % username
+
+
+def get_installation_metadata(event):
+    try:
+        integration = Integration.objects.get(external_id=event['installation']['id'])
+    except Integration.DoesNotExist:
+        return
+    return integration.metadata['installation']
+
+
+class Webhook(object):
```

Comment:
this looks really similar to the github one. can this be shared and just have a `provider` property that is used?

Reply:
yeah, actually, seems like with a lot of this, we could abstract out `provider` and maybe have a `get_client` and we could reuse more of the existing github code? maybe we can talk about that on monday?